app/controllers/api/v0/classified_listings_controller.rb

```diff
@@ -30,10 +31,13 @@ def index
       end
 
       def show
-        @classified_listing = ClassifiedListing.
-          select(ATTRIBUTES_FOR_SERIALIZATION).
-          includes(:user, :organization).
-          find(params[:id])
+        relation = ClassifiedListing.published
+
+        # if the user is authenticated we allow them to access
+        # their own unpublished listings as well
+        relation = relation.union(@user.classified_listings) if @user
+
+        @classified_listing = relation.select(ATTRIBUTES_FOR_SERIALIZATION).find(params[:id])
```

Comment:
We used to include `:user` and `:organization` associations, and now they are not preloaded. Is there a reason for that?

Reply:
See this conversation with @maestromac 

https://github.com/thepracticaldev/dev.to/pull/6458#discussion_r387906258

I tried with both and it doesn't seem to matter, as there are no N + 1 eager loading problems to solve. The queries are the same.

1. load the listings
1. load the user
1. load the org
1. load the tags

in both cases. 

I think the only way to eager load this is to use a single query with joins, which I tried with `          left_outer_joins(:user, :organization)` but it still results in 4 queries 🤷‍♂

Reply:
Let's keep this in mind, if the endpoint becomes slow we re-architect the queries